web/src/containers/Login.js

```diff
@@ -35,54 +36,39 @@ class Login extends Component {
       return <Redirect to="/" />;
     }
 
+    let errorMessage = false;
+    if (error === 'Unauthorized') {
+      errorMessage = 'The email or password you&apos;ve entered is incorrect.';
+    } else if (error === 'Unauthorized') {
+      errorMessage = 'Sorry! Something went wrong. Please try again.';
+    }
+
     return (
       <Fragment>
         <Header />
-        <div className="card--container">
-          <div className="ds-l-container card">
-            <div className="ds-l-row">
-              <div className="ds-l-col--1 ds-u-margin-left--auto" />
-              <div className="ds-l-col--12 ds-l-sm-col--10 ds-l-lg-col--6">
-                <h1 className="ds-h1">Log in</h1>
-                {error && error==="Unauthorized" && (
-                  <Alert variation='error' role='alert'>
-                    The email or password you&apos;ve entered is incorrect.
-                  </Alert>
-                )}
-                {error && error!=="Unauthorized" && (
-                  <Alert variation='error' role='alert'>
-                    <strong>Sorry!</strong> Something went wrong. Please try again.
-                  </Alert>
-                )}
-                <form onSubmit={this.handleSubmit}>
-                  <TextField
-                    id="username"
-                    label="Email"
-                    name="username"
-                    ariaLabel="Enter the email associated with this account."
-                    value={username}
-                    onChange={this.handleChange}
-                  />
-                  <Password
-                    title="Password"
-                    value={password}
-                    onChange={this.handleChange}
-                  />
-                  <Button
-                    type="submit"
-                    disabled={fetching || username.length === 0 || password.length === 0}
-                    variation="primary"
-                    className="ds-u-margin-y--4"
-                  >
-                  {fetching ? 'Logging in' : 'Log in'}
-                </Button>
-              </form>
-            </div>
-            <div className="ds-l-col--1 ds-u-margin-right--auto" />
-          </div>
-        </div>
-      </div>
-    </Fragment>
+
+        <CardForm
+          title="Log in"
+          error={errorMessage}
+          working={fetching}
+          primaryButtonText={['Log in', 'Logging in']}
+          onSave={this.handleSubmit}
+        >
+          <TextField
+            id="username"
+            label="Email"
+            name="username"
+            ariaLabel="Enter the email associated with this account."
+            value={username}
+            onChange={this.handleChange}
+          />
+          <Password
+            title="Password"
+            value={password}
+            onChange={this.handleChange}
+          />
+        </CardForm>
+      </Fragment>
```

Comment:
Look how much smaller it is now!

![](https://media.giphy.com/media/km2mais9qzYI/giphy.gif)

Reply:
So much cleaner!!!